sdks/python/apache_beam/runners/portability/fn_api_runner/fn_runner.py

```diff
@@ -471,9 +559,17 @@ def _collect_written_timers(
             timer_watermark_data[(transform_id, timer_family_id)] = min(
                 timer_watermark_data[(transform_id, timer_family_id)],
                 decoded_timer.hold_timestamp)
-        newly_set_timers[(transform_id, timer_family_id)] = ListBuffer(
-            coder_impl=timer_coder_impl)
-        newly_set_timers[(transform_id, timer_family_id)].append(out.get())
+          else:
+            # Timer was cleared, so we must skip setting it below.
+            timer_cleared = True
+            continue
+        if timer_cleared or (transform_id,
```

Comment:
what happens if same timer was set multiple times with clear being called in between?

Reply:
the SDK would only send back the latest of these events, independently of what it is. is that reasonable?

Reply:
I'm not sure that's the case when it is grpc data channel https://github.com/apache/beam/blob/master/sdks/python/apache_beam/runners/worker/bundle_processor.py#L669, IIUC every time timer calls set it will write an output timer to the output queue.

Reply:
ah my bad - only the last one will be saved - see in lines 547-548 of the file - we only store the latest event on a given tmier+window+key